Add Security Center table. closes #36 (#78)
turbot/steampipe-plugin-azure: docs/tables/azure_security_center.md

# Table: azure_security_center

Azure Security Center provides unified security management and advanced threat protection across hybrid cloud workloads. With Security Center, you can apply security policies across your workloads, limit your exposure to threats, and detect and respond to attacks.

## Examples

### Ensure that Microsoft Cloud App Security (MCAS) integration with Security Center is selected

```sql
select
  jsonb_pretty(setting) as setting
from
  azure_security_center
where
  jsonb_path_exists(
    setting,
    '$.** ? (@.type() == "string" && @ like_regex "MCAS")'
  );
```

### Ensure that Windows Defender ATP (WDATP) integration with Security Center is selected

```sql
select
  jsonb_pretty(setting) as setting
from
  azure_security_center
where
  jsonb_path_exists(
    setting,
    '$.** ? (@.type() == "string" && @ like_regex "WDATP")'
  );
```

### Check the status of the Automatic provisioning of monitoring agent

```sql
select
  jsonb_pretty(auto_provisioning) as auto_provisioning
from
  azure_security_center;

> ```

Reviewer: Please verify this query

### Ensure 'Additional email addresses' is configured with a security contact email

```sql
select
  jsonb_pretty(contact) as contact
from
  azure_security_center;

> ```

Reviewer: As per the query heading you should add a check for `email` field, otherwise it will always retrurn the whole contact object

# Table: azure_security_center

Azure Security Center provides unified security management and advanced threat protection across hybrid cloud workloads. With Security Center, you can apply security policies across your workloads, limit your exposure to threats, and detect and respond to attacks.

## Examples

### Ensure that Microsoft Cloud App Security (MCAS) integration with Security Center is selected

```sql
select
  jsonb_pretty(setting) as setting
from
  azure_security_center
where
  jsonb_path_exists(
    setting,
    '$.** ? (@.type() == "string" && @ like_regex "MCAS")'

>   );

Reviewer: Can we use `jsonb_array_elements`?

```

### Ensure that Windows Defender ATP (WDATP) integration with Security Center is selected

```sql
select
  jsonb_pretty(setting) as setting
from
  azure_security_center
where
  jsonb_path_exists(
    setting,
    '$.** ? (@.type() == "string" && @ like_regex "WDATP")'
  );
```

### Ensure that Automatic provisioning of monitoring agent is set to On

```sql
select

>   p -> 'properties' ->> 'autoProvision' as autoProvision

Reviewer: ```suggestion
  p -> 'properties' ->> 'autoProvision' as auto_provision
```

from
  azure_security_center,
  jsonb_array_elements(auto_provisioning) as p
where
  auto_provisioning is not null
  and p -> 'properties' ->> 'autoProvision' = 'On';
```

> ### Ensure 'Additional email addresses' is configured with a security contact email

Reviewer: ```suggestion
### Get security contact email configured for the subscription
```
